torch_xla/csrc/aten_xla_bridge.cpp

```diff
@@ -80,8 +80,12 @@ static absl::StatusOr<XLATensorImpl * absl_nonnull> GetXlaTensorImpl(
   XLATensorImpl* impl =
       dynamic_cast<XLATensorImpl*>(inner_tensor.unsafeGetTensorImpl());
   if (impl == nullptr) {
-    return XLA_ERROR_WITH_LOCATION(absl::InvalidArgumentError(absl::StrCat(
-        "Input tensor is not an XLA tensor: ", tensor.toString())));
+    auto error_message =
+        absl::StrCat("Failed retrieving the inner XLATensorImpl* from ",
```

Comment:
This error talks about C++ implementation details and isn't very actionable for users. How can a user trigger this error? Does it indicate a torch_xla bug? If so, we should crash instead.

Reply:
`GetXlaTensorImpl()` function is actually an internal function. My reasoning, here, was that: since it is an internal function, it should have somewhat more detailed information. Callers to this function (e.g. `GetXlaTensor()`, `GetInputXlaTensor()`) should overwrite the error message with something more appropriate to the end user.

> How can a user trigger this error?

Usually, when the underlying `TensorImpl` of a tensor (after unwrapping the functional layer) is not a `XLATensorImpl` instance. In other words, when the argument is not a proper XLA tensor.

> Does it indicate a torch_xla bug?

No. In this case, it indicates that this function failed.